### PR TITLE
Fixes next payment date appearing as 4th instead of 5th for a user in Australia on Android

### PR DESCRIPTION
**Pull request by tmancey**
Fixes https://github.com/brave/brave-browser/issues/7218

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

Confirm "Next payment date" is correct for different timezones around the world

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-confirmations/src/bat/confirmations/internal/payments.cc [hunk 1]
```diff
@@ -177,10 +177,14 @@ base::Time Payments::CalculateNextPaymentDate(
...
   }
 
-  auto date = base::StringPrintf("%d-%d-%d", year, month, kNextPaymentDay);
+  base::Time::Exploded next_payment_date_exploded = now_exploded;
+  next_payment_date_exploded.year = year;
+  next_payment_date_exploded.month = month;
+  next_payment_date_exploded.day_of_month = kNextPaymentDay;
 
   base::Time next_payment_date;
-  bool success = base::Time::FromString(date.c_str(), &next_payment_date);
```
Comment (tmancey) [inline vendor/bat-native-confirmations/src/bat/confirmations/internal/payments.cc]: FromString was working as expected on desktop, however, on Android, the dates were shown incorrectly

**Comment by NejcZdovc** [comment]
Can you please add unit test for it? Thank you

**Comment by tmancey** [comment]
@NejcZdovc Updated code and unit tests to use UTC and not Local time. As local time should only be used when displaying the time in UI

**Comment by tmancey** [comment]
Discussed with @mihaiplesa and unrelated CI failure

**Pull request merged**